pkg/ingester/ingester_ingest_storage_test.go

```diff
@@ -377,9 +527,33 @@ func TestIngester_ShouldNotCreatePartitionIfThereIsShutdownMarker(t *testing.T)
 	assert.Empty(t, watcher.PartitionRing().PartitionOwnerIDs(ingester.ingestPartitionID))
 }
 
-// Returned ingester and ring watcher are NOT started.
+func TestIngester_compactionServiceInterval(t *testing.T) {
+	cfg := defaultIngesterTestConfig(t)
+	overrides, err := validation.NewOverrides(defaultLimitsTestConfig(), nil)
+	require.NoError(t, err)
+	ingester, _, _ := createTestIngesterWithIngestStorage(t, &cfg, overrides, nil)
+
+	ingester.cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval = time.Minute
+	ingester.cfg.BlocksStorageConfig.TSDB.HeadCompactionIntervalJitterEnabled = false
+
+	firstInterval, standardInterval := ingester.compactionServiceInterval()
+	assert.Equal(t, time.Minute, firstInterval)
+	assert.Equal(t, time.Minute, standardInterval)
+
+	ingester.cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval = time.Minute
+	ingester.cfg.BlocksStorageConfig.TSDB.HeadCompactionIntervalJitterEnabled = true
+
+	firstInterval, standardInterval = ingester.compactionServiceInterval()
+	assert.Less(t, firstInterval, time.Minute)
+	assert.Equal(t, time.Minute, standardInterval)
+}
+
+// Returned ingester is NOT started.
 func createTestIngesterWithIngestStorage(t testing.TB, ingesterCfg *Config, overrides *validation.Overrides, reg prometheus.Registerer) (*Ingester, *kfake.Cluster, *ring.PartitionRingWatcher) {
-	defaultIngesterConfig := defaultIngesterTestConfig(t)
+	var (
```

Comment:
Ok. I personally find it easier to read, but I don't mind change my style (even if I would prefer if it's a linter telling me this styling stuff, so its consistency is enforced). That being said, this could be a topic we could discuss with the rest of the team to come up with a styling guide to follow.